EIPS/eip-6800.md

```diff
@@ -175,7 +175,12 @@ Bytes `1..4` are reserved for future use.
 
 Note: the code size is stored on 3 bytes. To allow for an extension to 4 bytes without changing the account version is possible, reserved byte #4 should be allocated last.
 
-When any account header field is set, the `version` field is also set to zero. The `code_keccak` and `code_size` fields are set upon contract or EoA creation.
+The `code_hash` and `code_size` fields are set upon contract or EoA creation.
+
+`version` can have two values:
+
+ * `0`, in which case this is a legacy contract. `code_hash` and `code_size` have the meaning they do in legacy contract;
+ * `1`, in which case this is an EOFv1 contract. The `code_size` field isn't set, and the `code_hash` leaf isn't created, as these values are known by the client.
```

Comment:
This has two consequences:

 * A new account version is created - if this is a problem, the bit can be set somewhere else
 * Values for code hash and size are not set. This is nice because it reduces the amount of computation required, as well as the costs. The drawback is that charging the gas costs for `EXTCODEHASH` for legacy contracts would become even more expensive: one would need to read the basic account data leaf in order to figure out if the code hash leaf should be read.

Reply:
I think we should encode in the spec how we differentiate EOFv1 contracts.  Specifically saying it is a contract starting with the two bytes `0xEF00` should be sufficient, and will keep design space open for EIP-7702's redirect markers.